src/ui/command.rs

```diff
@@ -415,13 +439,13 @@ pub fn spawn_input_thread(session: Session) -> thread::JoinHandle<()> {
                                     &mut tts_ctrl,
                                     &mut script,
                                 );
+                                writer
+                                    .send(Event::UserInputBuffer(
+                                        buffer.get_buffer(),
+                                        buffer.get_pos(),
+                                    ))
+                                    .unwrap();
                             }
-                            writer
-                                .send(Event::UserInputBuffer(
-                                    buffer.get_buffer(),
-                                    buffer.get_pos(),
-                                ))
-                                .unwrap();
```

Comment:
Quick note on this change: while testing I noticed that key binding events (e.g. F1, etc) would send a `UserInputBuffer` event even though none of the events `check_command_binds` processes could provide new input. This meant that if you set a prompt mask from a key binding it would immediately be invalidated when the unchanged prompt input is sent as a `UserInputBuffer` event.

Moving where the event was emitted so it is conditional on `check_command_binds` returning `false` fixed my test script so that I could press "f4" and have the highlights appear, where previously pressing "f4" would briefly flash the mask before it was invalidated and cleared:


```lua
blight.bind("f3", function ()
    blight.output("clearing mask");
    prompt_mask.clear()
end)
blight.bind("f4", function ()
  local data = prompt.get()
  if #data == 0 then
    blight.output("no prompt data")
    return
  end
  local danger_mask = {[1] = BG_RED, [#data+1] = C_RESET};
  local res = prompt_mask.set(data, danger_mask);
  blight.output(string.format("masked: %s",res))
end)
```

Reply:
Looks like you fixed a potential future bug in my eyes.